eth_utils/address.py

```diff
@@ -145,38 +135,42 @@ def is_normalized_address(value):
         return value == to_normalized_address(value)
 
 
-@coerce_args_to_bytes
-@coerce_return_to_bytes
-def to_canonical_address(address):
+def to_canonical_address(primitive=None, hexstr=None, text=None):
     """
+    Given any supported representation of an address
+    returns it's canonical form (20 byte long string).
     """
-    return decode_hex(to_normalized_address(address))
+    text_address = to_text(primitive, hexstr, text)
+    return decode_hex(to_normalized_address(text_address))
 
 
-def is_canonical_address(value):
-    if not is_address(value):
+def is_canonical_address(address):
+    """
+    Returns `True` if the `value` is an address in it's canonical form.
+    """
+    if not is_string(address) or not is_hex(address):
+        return False
+    elif not is_address(address):
         return False
     else:
-        return value == to_canonical_address(value)
+        return address == to_canonical_address(hexstr=address)
 
 
-@coerce_args_to_text
 def is_same_address(left, right):
     """
-    Checks if both addresses are same or not
+    Checks if both addresses are same or not.
     """
     if not is_address(left) or not is_address(right):
         raise ValueError("Both values must be valid addresses")
     else:
         return to_normalized_address(left) == to_normalized_address(right)
 
 
-@coerce_args_to_text
-@coerce_return_to_text
-def to_checksum_address(address):
+def to_checksum_address(primitive=None, hexstr=None, text=None):
```

Comment:
I don't think we can use the `primative/hexstr/text` signature here.  I actually think the original function *should* still work.  Is there something that breaks if this function is left unchanged?